insets/src/main/java/dev/chrisbanes/accompanist/insets/ImeNestedScrollConnection.kt

```diff
@@ -154,39 +156,41 @@ class ImeNestedScrollConnection(
         return Offset.Zero
     }
 
-    override fun onPostFling(
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override suspend fun onPostFling(
         consumed: Velocity,
-        available: Velocity,
-        onFinished: (Velocity) -> Unit
-    ) {
+        available: Velocity
+    ): Velocity {
         if (Build.VERSION.SDK_INT < 30) {
             // SimpleImeAnimationController only works on API 30+
-            onFinished(Velocity.Zero)
-            return
+            return Velocity.Zero
         }
 
         if (imeAnimController.isInsetAnimationInProgress()) {
             // If we have an IME animation in progress, from the user scrolling, we can
             // animate to the end state using the velocity
-            imeAnimController.animateToFinish(available.y) { remainingVelocity ->
-                onFinished(Velocity(x = 0f, y = remainingVelocity))
+            return suspendCancellableCoroutine { cont ->
+                imeAnimController.animateToFinish(available.y) { remainingVelocity ->
+                    cont.resume(Velocity(x = 0f, y = remainingVelocity)) {
+                        imeAnimController.cancel()
```

Comment:
Should this `invokeOnCancellation` as well?

Reply:
If not, use `suspendCoroutine`?

Reply:
That's what the resume block does https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-cancellable-continuation/resume.html

But maybe if you just use `.invokeOnCancellation`, you can avoid the experimental annotation

Reply:
Not quite. The `onCancellation` block on `resume()` is used for when the resume itself fails (due the outer coroutines being cancelled, etc). `invokeOnCancellation` is used for the normal cancelled state whilst suspended event.

Anyway, added `invokeOnCancellation`